### PR TITLE
Added a note about N+1 problem and form collections

### LINKED ISSUE #3374: Add cookbook article about overriding predefined data for child form
https://github.com/symfony/symfony/issues/9746

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #3374 and #4323.

### CODE REVIEW

File: reference/forms/types/entity.rst [hunk 1]
```diff
@@ -99,6 +99,13 @@ the `query_builder`_ option::
         'choice_label' => 'username',
     ]);
 
+.. note::
+
+    Using form collections may result in making too many database requests to
+    fetch related entities. This is know as the *"N + 1 query problem"* and it
```
Comment (OskarStark) [inline reference/forms/types/entity.rst]: s/know/known/

File: reference/forms/types/entity.rst [hunk 2]
```diff
@@ -99,6 +99,13 @@ the `query_builder`_ option::
...
+    can be solved :ref:`joining related records <doctrine-associations-join-query>`
```
Comment (OskarStark) [inline reference/forms/types/entity.rst]: s/solved/solved by/ ?

**Review by OskarStark — CHANGES_REQUESTED** [review]

**Comment by wouterj** [comment]
Please note that while joining seems like a performance optimalization, it makes hydration of doctrine way more complex for *..n relations. This often reduces performance instead of optimizing it: https://tideways.com/profiler/blog/5-doctrine-orm-performance-traps-you-should-avoid

**Comment by javiereguiluz** [comment]
@wouterj I read the article you linked, but decided to merge this anyways. You can see that even the article contradicts itself because in (2) they recommend to not use joins ... but in (5) they recommend to use joins.

This is a tricky issue ... but at least we now display a warning message about this so people can investigate on their own. Thanks.

**Pull request closed without merging**